src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts

```diff
@@ -0,0 +1,529 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useLocation, useSearchParams } from "react-router-dom";
+import { ROWS_PER_PAGE_DEFAULT_OPTIONS } from "../../Components/Filters/NumberOfRowsFilter";
+import { ContentTypeBase, SortDirection, TableColumn } from "../../types";
+import { useDebounce } from "./useDebounce";
+
+enum FilteredTableDataQueryParam {
+  SortBy = "sortBy",
+  Order = "order",
+  Query = "query",
+  ContentTypeBase = "type",
+  ShowColumn = "showColumn",
+  Page = "page",
+  RowsPerPage = "rowsPerPage",
+}
+
+interface FilteredTableDataHookOptions<TableDataType> {
+  rows: TableDataType[];
+  initialTableColumns: TableColumn<TableDataType>[];
+  initialContentTypeBases?: ContentTypeBase[];
+  contentTypeBaseColumnId?: string;
```

Comment:
Can contentTypeBaseColumnId be different than FilteredTableDataQueryParam.ContentTypeBase? Maybe it's worth to change string to FilteredTableDataQueryParam.ContentTypeBase.
```suggestion
  contentTypeBaseColumnId?: FilteredTableDataQueryParam.ContentTypeBase;
```